src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -237,10 +269,129 @@ public Boolean approveTeam(Set<Player> players) {
+                .filter(player -> player != null && player.getPosition().equals("Attacker"))
+                .count();
+
+
+        long startingGoalkeepers = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Goalkeeper"))
+                .count();
+
+        if (position.equals("Goalkeeper") && startingGoalkeepers >= MIN_GK) {
+            throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 1 goalkeeper in the starting lineup");
+
+        } else if (!position.equals("Goalkeeper") && startingGoalkeepers < MIN_GK) { // might have to delete this condition
+            throw new Error400("Fantasy team with id " + fantasyTeamId + " needs at least 1 goalkeeper in the starting lineup");
+
+        } else if (position.equals("Defender")) {
+            if (totalDefenders > TOTAL_DEF) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 9 defenders in total");
+            } else if (startingDefenders > MAX_DEF) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 5 defenders in the starting lineup");
+            }
+
+        } else if (position.equals("Midfielder")) {
+            if (totalMidfielders > TOTAL_MID) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 9 midfielders in total");
+            } else if (startingMidfielders > MAX_MID) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 5 midfielders in the starting lineup");
+            }
+
+        } else if (position.equals("Attacker")) {
+            if (totalAttackers > TOTAL_ATT) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 7 attackers in total");
+            } else if (startingAttackers > MAX_ATT) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 3 attackers in the starting lineup");
+            }
+        }
+        return true;
+    }
+
+    /*
+    * This method substitutes the incoming player on the first eligible bench player in the fantasy team
+     */
+    private boolean substitutePlayer(FantasyTeam proposingFantasyTeam, Player outgoingPlayer, String[] benchPlayerIds) {
+        for (String benchPlayerId : benchPlayerIds) {
+            //check if we exceed max number of players per position if we were to pick that bench player
+            if (checkExcedentPositions(proposingFantasyTeam, playerService.getPlayerById(Long.valueOf(benchPlayerId)).getPosition())) {
+                proposingFantasyTeam.setPlayerIdsInOrder(changeLineupString(proposingFantasyTeam.getPlayerIdsInOrder(), benchPlayerId, outgoingPlayer.getPlayerId().toString()));
```

Comment:
This wont work because the string after changeLineupString will have the benchPlayer still on the bench and in outgoing players position